aten/src/ATen/native/cuda/RangeFactories.cu

```diff
@@ -49,102 +49,118 @@ void gpu_kernel_with_index(at::Tensor &output, func_t f) {
+      "throw a runtime error in a future release. This warning will appear "
+      "only once per process.");
+  }
 
   if (result.numel() != steps) {
-    result.resize_({steps});
+    result.resize_({steps_});
   }
   bool is_contiguous = result.is_contiguous();
   Tensor r = !is_contiguous ? at::empty_like(result, LEGACY_CONTIGUOUS_MEMORY_FORMAT) : result;
 
-  if (steps == 0) {
+  if (steps_ == 0) {
     // skip
-  } else if (steps == 1) {
+  } else if (steps_ == 1) {
     r.fill_(start);
   } else if (isIntegralType(r.scalar_type(), 0)) {
     AT_DISPATCH_INTEGRAL_TYPES(r.scalar_type(), "linspace_cuda", [&]() {
       scalar_t scalar_start = start.to<scalar_t>();
       scalar_t scalar_end = end.to<scalar_t>();
       // Cast `end` and `start` to `float`, since range can be larger than scalar_t for integral types
-      float step = (static_cast<float>(scalar_end) - static_cast<float>(scalar_start)) / (steps - 1);
-      const int64_t halfway = steps / 2;
-      gpu_kernel_with_index(r, [scalar_start, scalar_end, steps, step, halfway]GPU_LAMBDA(int64_t ind) -> scalar_t {
+      float step = (static_cast<float>(scalar_end) - static_cast<float>(scalar_start)) / (steps_ - 1);
+      const int64_t halfway = steps_ / 2;
+      gpu_kernel_with_index(r, [scalar_start, scalar_end, steps_, step, halfway]GPU_LAMBDA(int64_t ind) -> scalar_t {
         if (ind < halfway) {
           return scalar_start + (step * ind);
         }
 
-        return scalar_end - step * (steps - ind - 1);
+        return scalar_end - step * (steps_ - ind - 1);
       });
     });
   } else {
     AT_DISPATCH_FLOATING_TYPES_AND_HALF(r.scalar_type(), "linspace_cuda", [&]() {
       scalar_t scalar_start = start.to<scalar_t>();
       scalar_t scalar_end = end.to<scalar_t>();
-      scalar_t step = (scalar_end - scalar_start) / static_cast<scalar_t>(steps - 1);
-      const int64_t halfway = steps / 2;
-      gpu_kernel_with_index(r, [scalar_start, scalar_end, steps, step, halfway]GPU_LAMBDA(int64_t ind) -> scalar_t {
+      scalar_t step = (scalar_end - scalar_start) / static_cast<scalar_t>(steps_ - 1);
+      const int64_t halfway = steps_ / 2;
+      gpu_kernel_with_index(r, [scalar_start, scalar_end, steps_, step, halfway]GPU_LAMBDA(int64_t ind) -> scalar_t {
         if (ind < halfway) {
           return scalar_start + (step * ind);
         }
 
-        return scalar_end - step * (steps - ind - 1);
+        return scalar_end - step * (steps_ - ind - 1);
       });
     });
   }
 
-  if(!is_contiguous) {
+  if (!is_contiguous) {
     result.copy_(r);
   }
 
   AT_CUDA_CHECK(cudaGetLastError());
   return result;
 }
 
-Tensor& logspace_cuda_out(Tensor& result, Scalar start, Scalar end, int64_t steps, double base) {
-  TORCH_CHECK(steps >= 0, "number of steps must be non-negative");
+Tensor& logspace_cuda_out(Tensor& result, Scalar start, Scalar end, c10::optional<int64_t> steps, double base) {
+  const auto steps_ = steps.value_or(100);
```

Comment:
I've definitely done this before and regretted it when I came back and started using "steps" and forgot I need to check if it has a value.  I'd give the parameter the weird name (could even be _steps).  It's unfortunate it won't match up with the forward declare but such is life.

Reply:
I like your suggestion that `_steps` is weirder than `steps_`. What about making this `stps` so `steps` is neither a prefix nor suffix of the variable name?

Reply:
Offline: update the "steps" argument to be "steps_optional" and the derived value to be "steps"